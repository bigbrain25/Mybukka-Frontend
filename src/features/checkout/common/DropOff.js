import React from 'react';
import { useHistory } from 'react-router-dom';

import TemporaryWrapper from '../../../components/ViewWrappers/TemporaryWrapper';

import Address from '../../../components/address/Address';
import GeoSuggestions from '../../../components/places-suggest/GeoSuggestions';
import { useUserContext } from '../../../context/UserContext';


const DropOff = () => {
  const { address: addresses } = useUserContext();
  const { push } = useHistory();

  return (
    <TemporaryWrapper>
      {(!addresses || !addresses.addresses.length) && <Address withFormSpace withPadding label="Delivery Address" />}
      <div className="addresses-section">
        {addresses && addresses.addresses.map(({ address, slug, _id: id }) => (
          addresses.defaultAddress === slug ?
            <GeoSuggestions
              asUtility
              handleClick={() => {}}
              text="Change"
              withPrimaryButton
              emitOnClick={() => push('/profile#addresses')}
              predictions={[{ terms: address.split(', ').map(loc => ({ value: loc })) }]}
              key={`Plain-Account-Details-DELETE-$${id}`}
            />
            : null
        ))}
      </div>
    </TemporaryWrapper>
  );
};

export default DropOff;
