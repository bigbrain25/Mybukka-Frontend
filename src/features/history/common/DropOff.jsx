import React from 'react';
import PropTypes from 'prop-types';

const DropOff = ({ deliveryName, deliveryAddress, deliveryContactMobile }) => (
  <div className="dropoff">
    <h6 className="font-size-16">{deliveryName}</h6>
    <h6 className="font-size-14">{deliveryAddress}</h6>
    <h6 className="font-size-14">{deliveryContactMobile}</h6>
  </div>
);

export default DropOff;

DropOff.propTypes = {
  deliveryName: PropTypes.string.isRequired,
  deliveryAddress: PropTypes.string.isRequired,
  deliveryContactMobile: PropTypes.string.isRequired
};
