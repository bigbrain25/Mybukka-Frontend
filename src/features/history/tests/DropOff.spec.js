import React from 'react';
import DropOff from '../common/DropOff';

describe('DropOff component', () => {
  const props = {
    deliveryName: 'delivery name',
    deliveryAddress: 'delivery address',
    deliveryContactMobile: '09098891282',
  };

  const wrapper = mount(<DropOff {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
