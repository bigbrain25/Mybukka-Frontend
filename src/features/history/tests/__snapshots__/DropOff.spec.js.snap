// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropOff component renders properly 1`] = `ReactWrapper {}`;
